tests/test_declaration.py

```diff
@@ -443,6 +447,111 @@ def test_directional_property(StyleClass):
     )
 
 
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        ([VALUE1], [VALUE1]),
+        ([VALUE1, VALUE3], [VALUE1, VALUE3]),
+        ([VALUE2, VALUE1], [VALUE2, VALUE1]),
+        ([VALUE2, VALUE3, 1, 2, VALUE1], [VALUE2, VALUE3, 1, 2, VALUE1]),
+        # Duplicates are kept, but "normalized" via validation.
+        (
+            [VALUE3, 1, VALUE3, "1", True, " 1", VALUE2],
+            [VALUE3, 1, VALUE3, 1, 1, 1, VALUE2],
+        ),
+        # Other sequences should work too.
+        ((VALUE1, VALUE3), [VALUE1, VALUE3]),
+    ],
+)
+def test_list_property(value, expected):
+    style = Style()
+    style.list_prop = value
+    assert style.list_prop == expected
+
+
+@pytest.mark.parametrize(
+    "value, error, match",
+    [
+        (
+            VALUE2,
+            TypeError,
+            r"Value for list property list_prop must be a non-string sequence\.",
+        ),
+        (
+            5,
+            TypeError,
+            r"Value for list property list_prop must be a non-string sequence\.",
+        ),
+        (
+            # Fails because it's only a generator, not a comprehension:
+            (i for i in [VALUE1, VALUE3]),
+            TypeError,
+            r"Value for list property list_prop must be a non-string sequence.",
+        ),
+        (
+            [VALUE3, VALUE1, "bogus"],
+            ValueError,
+            r"Invalid value 'bogus' for property list_prop; Valid values are: "
+            r"none, value1, value2, value3, <integer>",
+        ),
+        (
+            (),
+            ValueError,
+            r"Series properties cannot be set to an empty sequence; "
+            r"to reset a property, use del `style.list_prop`\.",
+        ),
+        (
+            [],
+            ValueError,
+            r"Series properties cannot be set to an empty sequence; "
+            r"to reset a property, use del `style.list_prop`\.",
+        ),
+    ],
+)
+def test_list_property_invalid(value, error, match):
+    style = Style()
+    with pytest.raises(error, match=match):
+        style.list_prop = value
+
+
+@pytest.mark.parametrize(
+    "attr",
+    [
+        "__set_item__",
+        "__del_item",
+        "insert",
+        "append",
+        "clear",
+        "reverse",
+        "extend",
+        "pop",
+        "remove",
+        "__iadd__",
+        "sort",
+    ],
+)
+def test_list_property_immutable(attr):
+    style = Style()
+    style.list_prop = [1, 2, 3, VALUE2]
+    with pytest.raises(AttributeError):
+        getattr(style.list_prop, attr)
```

Comment:
This checks all the "list-like" methods, but doesn't check direct usage - I'd argue it would be better to exercise those directly, as it's more explicit (and it avoids the "wrong method name still raises an AttributeError" potential for error).